### PR TITLE
Added 4 Gen2 Network TC

### PR DESCRIPTION
**Pull request by asofron**
Test case names :
Hot-Add Multiple NICs - Gen2 VM
Hot-Remove Multiple NICs - Gen2 VM
Boot no NIC Hot-Add NIC - Gen2 VM
Maximum synthetic NICs hot-add
Maximum synthetic NICs hot-remove

 VHD under test **ubuntu_18.04_x64_gen2vm.vhdx**
 ARM Image under test  :  :  : 
 Total Executed TestCases        4 (       4 Pass,        0 Fail,        0 Abort)
 Total Execution Time(dd:hh:mm) 0:0:9
 XML file: TestConfiguration
 
 #Sr. Test Name : Test Result : Test Duration (in minutes)
 ----------------------------------------------------
 1. NET-HOT-ADD-MULTINIC : PASS : 2.13
 2. NET-HOT-REMOVE-MULTINIC : PASS : 0.64
 3. NET-BOOT-NONIC-HOT-ADD-NIC : PASS : 2.29
 4. NET-HOT-ADD-REMOVE-MAXNIC : PASS : 4.06

### CODE REVIEW

File: Libraries/CommonFunctions.psm1 [hunk 1]
```diff
@@ -4284,4 +4284,106 @@ Function Remove-InvalidCharactersFromFileName
...
     $Regex = "[{0}]" -f [RegEx]::Escape($WindowsInvalidCharacters)
     return ($FileName -replace $Regex)
+}
+
+function Get-KVPItem {
+	param (
+	    $VMName,
+	    $server,
+	    $keyName,
+		$Intrinsic
```
Comment (adriansuhov) [inline Libraries/CommonFunctions.psm1]: indentation

File: Libraries/CommonFunctions.psm1 [hunk 2]
```diff
@@ -4284,4 +4284,106 @@ Function Remove-InvalidCharactersFromFileName
...
+	)
+
+    $vm = Get-WmiObject -Namespace root\virtualization\v2 -Query "Select * From Msvm_ComputerSystem Where ElementName=`'$VMName`'"
+    if (-not $vm)
+    {
```
Comment (adriansuhov) [inline Libraries/CommonFunctions.psm1:4299]: I know it means changes in a lot of places, but please stick with 
"if (check) {
} else {
}"

File: Libraries/CommonFunctions.psm1 [hunk 3]
```diff
@@ -4284,4 +4284,106 @@ Function Remove-InvalidCharactersFromFileName
...
+        {
+            return $value
+        }
+    }
+
+    return $Null
+}
+
+function Add-RemoveMaxNIC {
+	param(
```
Comment (adriansuhov) [inline Libraries/CommonFunctions.psm1]: Remove double indentation in the entire function Add-RemoveMaxNIC

File: Libraries/CommonFunctions.psm1 [hunk 4]
```diff
@@ -4284,4 +4284,106 @@ Function Remove-InvalidCharactersFromFileName
...
+		$actionType,
+		[int] $nicsAmount
+	)
+	for ($i=1; $i -le $nicsAmount; $i++)
+	{
+		$nicName = "External" + $i
+
+		if ($actionType -eq "add")
+		{
+			"Info : Ensure the VM does not have a Synthetic NIC with the name '${nicName}'"
```
Comment (adriansuhov) [inline Libraries/CommonFunctions.psm1]: LogMsg "Ensure ... " (use LogMsg and remove "Info : " from the message). There are other occurrences

File: Testscripts/Linux/NET-Verify-Boot-NoNIC.sh [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+chmod 755 ./kvp_client64
+
+# Verify there are no eth devices
+echo "Info : Check count of eth devices"
+#ethCount=$(ifconfig -a | grep '^eth' | wc -l)
+ethCount=$(ls -d /sys/class/net/eth* | wc -l)
+echo "Info : ethCount = ${ethCount}"
+if [ $ethCount -ne 0 ]; then
+    echo "eth device count is not zero: ${ethCount}"
+
```
Comment (adriansuhov) [inline Testscripts/Linux/NET-Verify-Boot-NoNIC.sh]: remove new line

File: Testscripts/Linux/NET-Verify-Boot-NoNIC.sh [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    echo "the file kvp_client64 does not exist"
+    exit 0
+fi
+
+echo "Info : chmod 755 ./kvp_client64"
+chmod 755 ./kvp_client64
+
+# Verify there are no eth devices
+echo "Info : Check count of eth devices"
+#ethCount=$(ifconfig -a | grep '^eth' | wc -l)
```
Comment (adriansuhov) [inline Testscripts/Linux/NET-Verify-Boot-NoNIC.sh]: remove comment

File: Testscripts/Linux/NET-Verify-Boot-NoNIC.sh [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+ethCount=$(ls -d /sys/class/net/eth* | wc -l)
+echo "Info : ethCount = ${ethCount}"
+if [ $ethCount -ne 0 ]; then
+    echo "eth device count is not zero: ${ethCount}"
+
+    exit 0
+fi
+
```
Comment (adriansuhov) [inline Testscripts/Linux/NET-Verify-Boot-NoNIC.sh:21]: double new line

File: Testscripts/Linux/NET-Verify-Boot-NoNIC.sh [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+    fi
+
+    timeout=$((timeout-10))
+    sleep 10
+    if [ $timeout -le 0 ]; then
+        echo "Timed out waiting for eth device to be created"
+        exit 0
+    fi
+done
+
```
Comment (adriansuhov) [inline Testscripts/Linux/NET-Verify-Boot-NoNIC.sh:45]: double new line

File: Testscripts/Linux/NET-Verify-HotAdd-MultiNIC.sh [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+        SetTestStateAborted
+        exit 0
+    fi
+}
+
+# Determine how many eth devices the OS sees
+eth_count=$(ls -d /sys/class/net/eth* | wc -l)
+LogMsg "eth_count = ${eth_count}"
+
+
```
Comment (adriansuhov) [inline Testscripts/Linux/NET-Verify-HotAdd-MultiNIC.sh:116]: double new line

File: Testscripts/Linux/NET-Verify-HotAdd-MultiNIC.sh [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+# Licensed under the Apache License.
+
+. utils.sh || {
+    echo "Error: unable to source utils.sh!"
+    exit 0
+}
+
+# Source constants file and initialize most common variables
+UtilsInit
+
```
Comment (adriansuhov) [inline Testscripts/Linux/NET-Verify-HotAdd-MultiNIC.sh:13]: double new line

File: Testscripts/Windows/NET-Boot-NoNIC-HotAdd-NIC.ps1 [hunk 11]
```diff
@@ -0,0 +1,243 @@
...
+        }
+
+        $tmo -= 10
+        Start-Sleep -Seconds 10
+    }
+
+    if ($value -ne "NoNICs") {
+        LogErr "The VM never detected the Hot Remove of the NIC"
+        return "FAIL"
+    }
```
Comment (adriansuhov) [inline Testscripts/Windows/NET-Boot-NoNIC-HotAdd-NIC.ps1:234]: Where is the 'return "PASS"'?

File: Testscripts/Windows/NET-HotAdd-MultiNIC.ps1 [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+    RemoteCopy -download -downloadFrom $Ipv4 -files "/home/${VMUserName}/NET-Verify-${nic_action}-MultiNIC.log" `
+        -downloadTo $LogDir -port $VMPort -username $VMUserName -password $VMPassword
+    $contents = Get-Content -Path $stateFile
+    if (($contents -eq "TestAborted") -or ($contents -eq "TestFailed")) {
+        LogErr "Failed to verify the VM detected the hot ${nic_action} !"
+        return "FAIL"
+    }
+    else {
+        LogMsg "Test PASSED,VM detected the hot ${nic_action} !"
+    }
```
Comment (adriansuhov) [inline Testscripts/Windows/NET-HotAdd-MultiNIC.ps1:110]: Where is the 'return "PASS"'?

File: Testscripts/Windows/NET-HotAddRemove-Max-NIC.ps1 [hunk 13]
```diff
@@ -0,0 +1,143 @@
...
+
+    # Check if KVP IP values match the ones present in the VM
+    LogMsg "Info : Checking KVP values for each NIC"
+    Start-Sleep -s 60
+    $kvp_ip = Get-IPv4ViaKVP $vmName $hvServer | Select-Object -uniq
+    $vm_ip = RunLinuxCmd -username $VMUserName -password $VMPassword -ip $Ipv4 -port $VMPort -command "ip -4 -o addr show scope global | awk '{print `$4}'" | ForEach-Object {$_.Split('\n')} | ForEach-Object { $_.Split('/')[0]; }
+
+    if ($kvp_ip.length -ne $vm_ip.length) {
+        LogErr "IP values sent through KVP are not the same as the ones from the VM"
+        LogErr	"KVP values : ${kvp_ip} VM values : ${vm_ip}"
```
Comment (adriansuhov) [inline Testscripts/Windows/NET-HotAddRemove-Max-NIC.ps1]: Unnecessary spaces

File: Testscripts/Windows/NET-HotAddRemove-Max-NIC.ps1 [hunk 14]
```diff
@@ -0,0 +1,143 @@
...
+    if ($osInfo -le 9600) {
+        LogErr  "This test requires Windows Server 2016 or higher"
+        return "ABORTED"
+    }
+
+
+    #	Hot Add maximum number of synthetic NICs
+    LogMsg "Hot Adding the maximum number of synthetic NICs ..."
+    $addnic = Add-RemoveMaxNIC $vmName $hvServer $switch_name "add" $nicsAmount
+
```
Comment (adriansuhov) [inline Testscripts/Windows/NET-HotAddRemove-Max-NIC.ps1:70]: double new line

File: Testscripts/Windows/NET-HotAddRemove-Max-NIC.ps1 [hunk 15]
```diff
@@ -0,0 +1,143 @@
...
+    $osInfo = Get-HostBuildNumber $hvServer
+    if (-not $osInfo) {
+        LogErr "Unable to collect Operating System information"
+        return "ABORTED"
+    }
+    if ($osInfo -le 9600) {
+        LogErr  "This test requires Windows Server 2016 or higher"
+        return "ABORTED"
+    }
+
```
Comment (adriansuhov) [inline Testscripts/Windows/NET-HotAddRemove-Max-NIC.ps1:66]: double new line

File: Testscripts/Windows/NET-HotAddRemove-Max-NIC.ps1 [hunk 16]
```diff
@@ -0,0 +1,143 @@
...
+        return "FAIL"
+    }
+    if ($contents2 -match "netvsc throwed errors") {
+        LogErr "Error: VM '${vmName}' reported that netvsc throwed errors"
+        return "FAIL"
+    }
+    else {
+        LogMsg "Test PASSED ,VM detected the hot remove!"
+    }
+ LogMsg "Mutiple NICs Hot Add/Remove successfully..."
```
Comment (adriansuhov) [inline Testscripts/Windows/NET-HotAddRemove-Max-NIC.ps1]: Indentation

**Review by adriansuhov — CHANGES_REQUESTED** [review]

File: Libraries/CommonFunctions.psm1 [hunk 17]
```diff
@@ -4284,4 +4284,106 @@ Function Remove-InvalidCharactersFromFileName
...
+			{
+				Throw "Error: VM '${vmName}' already has a NIC named '${nicName}'"
+			}
+		}
+
+		"Info : Hot '${actionType}' a synthetic NIC with name of '${nicName}' using switch '${switchName}'"
+		"Info : Hot '${actionType}' '${network_type}' to '${vmName}'"
+		if ($actionType -eq "add")
+		{
+			Add-VMNetworkAdapter -VMName $vmName -SwitchName $network_type -ComputerName $hvServer -Name ${nicName} #-ErrorAction SilentlyContinue
```
Comment (chvalean) [inline Libraries/CommonFunctions.psm1]: remove comment at end of line if not required, but maybe keep the ErrorAction - not as a comment

File: Testscripts/Linux/NET-Verify-Boot-NoNIC.sh [hunk 18]
```diff
@@ -0,0 +1,92 @@
...
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
+
+if [ ! -e ./kvp_client64 ]; then
+    echo "the file kvp_client64 does not exist"
+    exit 0
+fi
+
+echo "Info : chmod 755 ./kvp_client64"
```
Comment (chvalean) [inline Testscripts/Linux/NET-Verify-Boot-NoNIC.sh]: no need to display this

File: Testscripts/Linux/NET-Verify-HotAdd-MultiNIC.sh [hunk 19]
```diff
@@ -0,0 +1,137 @@
...
+            sed -i -e "s/HWADDR/#HWADDR/" /etc/sysconfig/network-scripts/ifcfg-${eth_name}
+            sed -i -e "s/UUID/#UUID/" /etc/sysconfig/network-scripts/ifcfg-${eth_name}
+    elif [ "$os_vendor" == "SUSE LINUX" ] || \
+	     [ "$os_vendor" == "SLE" ]; then
+            LogMsg "Info : Creating ifcfg-${eth_name}"
+            cp /etc/sysconfig/network/ifcfg-eth0 /etc/sysconfig/network/ifcfg-${eth_name}
+            sed -i -- "s/eth0/${eth_name}/g" /etc/sysconfig/network/ifcfg-${eth_name}
+            sed -i -e "s/HWADDR/#HWADDR/" /etc/sysconfig/network/ifcfg-${eth_name}
+            sed -i -e "s/UUID/#UUID/" /etc/sysconfig/network/ifcfg-${eth_name}
+    elif [ "$os_vendor" == "Ubuntu" ]; then
```
Comment (chvalean) [inline Testscripts/Linux/NET-Verify-HotAdd-MultiNIC.sh]: add debian as well

File: Testscripts/Linux/NET-Verify-HotAdd-MultiNIC.sh [hunk 20]
```diff
@@ -0,0 +1,137 @@
...
+            echo "auto ${eth_name}" >> /etc/network/interfaces
+            echo "iface ${eth_name} inet dhcp" >> /etc/network/interfaces
+    else
+        LogErr "Error: Linux Distro not supported!"
```
Comment (chvalean) [inline Testscripts/Linux/NET-Verify-HotAdd-MultiNIC.sh]: remove 'Error:'

File: Testscripts/Windows/NET-Boot-NoNIC-HotAdd-NIC.ps1 [hunk 21]
```diff
@@ -0,0 +1,243 @@
...
+    )
+
+    $KVP_KEY = "HotAddTest"
+    $Switch_Name = "External"
+
+    # Change the working directory to where we should be
+    Set-Location $RootDir
+
+    # Verify the target VM exists, and that it is a Gen 2 VM
+    "Info : Verify the SUT VM '${VMName}' exists"
```
Comment (chvalean) [inline Testscripts/Windows/NET-Boot-NoNIC-HotAdd-NIC.ps1]: remove this check

File: Testscripts/Windows/NET-Boot-NoNIC-HotAdd-NIC.ps1 [hunk 22]
```diff
@@ -0,0 +1,243 @@
...
+    $vm = Get-VM -Name $VMName -ComputerName $HvServer -ErrorAction SilentlyContinue
+    if (-not $?) {
+        LogErr "Unable to find VM '${VMName}' on server '${HvServer}'"
+    }
+
+    if ($vm.Generation -ne 2) {
+        LogWarn "This test requires a Gen 2 VM.  VM '${VMName}' is not a Gen2 VM"
```
Comment (chvalean) [inline Testscripts/Windows/NET-Boot-NoNIC-HotAdd-NIC.ps1]: LogErr  and rephrase this better, required a gen2, is not gen2.
Keep consistent name "Gen2 VM"

File: Testscripts/Windows/NET-Boot-NoNIC-HotAdd-NIC.ps1 [hunk 23]
```diff
@@ -0,0 +1,243 @@
...
+    $linuxRelease = DetectLinuxDistro
+    if ($linuxRelease -eq "CENTOS" -or $linuxRelease -eq "FEDORA" -or $linuxRelease -eq "RHEL") {
+        RunLinuxCmd -username $VMUserName -password $VMPassword -ip $Ipv4 -port $VMPort -command `
+            "chmod +x /etc/rc.d/rc.local" -runAsSudo
+        RunLinuxCmd -username $VMUserName -password $VMPassword -ip $Ipv4 -port $VMPort -command `
+            "echo 'chmod 755 /home/$VMUserName/NET-Verify-Boot-NoNIC.sh' >> /etc/rc.d/rc.local" -runAsSudo
+        RunLinuxCmd -username $VMUserName -password $VMPassword -ip $Ipv4 -port $VMPort -command `
+            "echo 'cd /home/$VMUserName/ && ./NET-Verify-Boot-NoNIC.sh > /home/$VMUserName/NET-Verify-Boot-NoNIC.log  &' >> /etc/rc.d/rc.local" -runAsSudo
+    }
+    elseif ($linuxRelease -eq "UBUNTU") {
```
Comment (chvalean) [inline Testscripts/Windows/NET-Boot-NoNIC-HotAdd-NIC.ps1]: or Debian

File: Testscripts/Windows/NET-Boot-NoNIC-HotAdd-NIC.ps1 [hunk 24]
```diff
@@ -0,0 +1,243 @@
...
+            "echo '$cmdToVM' > /etc/systemd/system/after-local.service" -runAsSudo
+        RunLinuxCmd -username $VMUserName -password $VMPassword -ip $Ipv4 -port $VMPort -command `
+            "chmod +x /etc/systemd/system/after-local.service" -runAsSudo
+        RunLinuxCmd -username $VMUserName -password $VMPassword -ip $Ipv4 -port $VMPort -command `
+            "chmod 755 /home/$VMUserName/NET-Verify-Boot-NoNIC.sh" -runAsSudo
+        RunLinuxCmd -username $VMUserName -password $VMPassword -ip $Ipv4 -port $VMPort -command `
+            "systemctl daemon-reload && systemctl enable after-local.service" -runAsSudo
+    }
+    else {
+        LogErr "Unknown linux distribution '${linuxRelease}'"
```
Comment (chvalean) [inline Testscripts/Windows/NET-Boot-NoNIC-HotAdd-NIC.ps1]: Unsupported*

File: Testscripts/Windows/NET-Boot-NoNIC-HotAdd-NIC.ps1 [hunk 25]
```diff
@@ -0,0 +1,243 @@
...
+    if (-not $?) {
+        LogErr "Unable to stop VM to allow removal of original NIC"
+        return "FAIL"
+    }
+
+    # Remove the original NIC
+    LogMsg "Removing the original NIC from the VM..."
+    Remove-VMNetworkAdapter -VMName $VMName -ComputerName $HvServer -ErrorAction SilentlyContinue
+    if (-not $?) {
+        LogErr  "Unable to Hot Remove NIC"
```
Comment (chvalean) [inline Testscripts/Windows/NET-Boot-NoNIC-HotAdd-NIC.ps1]: this is not hot remove, the VM gets stopped above

File: Testscripts/Windows/NET-Boot-NoNIC-HotAdd-NIC.ps1 [hunk 26]
```diff
@@ -0,0 +1,243 @@
...
+        return "FAIL"
+    }
+
+    LogMsg "Verifing the VM does not have any NICs..."
```
Comment (chvalean) [inline Testscripts/Windows/NET-Boot-NoNIC-HotAdd-NIC.ps1]: Verifying*

File: Testscripts/Windows/NET-HotAdd-MultiNIC.ps1 [hunk 27]
```diff
@@ -0,0 +1,120 @@
...
+            default {}  # unknown param - just ignore it
+        }
+    }
+
+    # Change the working directory to where we should be
+    LogMsg "Changing directory to '${rootDir}'"
+    Set-Location $RootDir
+
+    # Verify the target VM exists, and that it is a Gen2 VM
+    LogMsg "Verifing the target VM exists, and that it is a Gen2 VM.."
```
Comment (chvalean) [inline Testscripts/Windows/NET-HotAdd-MultiNIC.ps1]: remove this check

File: Testscripts/Windows/NET-HotAddRemove-Max-NIC.ps1 [hunk 28]
```diff
@@ -0,0 +1,143 @@
...
+        $TestParams
+    )
+
+    $switch_name = "External"
+    $nicsAmount = "7"
+    $remoteScript = "NET-MAX-NICs.sh"
+    $remoteScript2 = "NET-Verify-HotAdd-MultiNIC.sh"
+    ########################################################################
+
+    # Display the test parameters in the log file
```
Comment (chvalean) [inline Testscripts/Windows/NET-HotAddRemove-Max-NIC.ps1]: remove the display of params

File: Testscripts/Windows/NET-HotAddRemove-Max-NIC.ps1 [hunk 29]
```diff
@@ -0,0 +1,143 @@
...
+    LogMsg "Test parameters:"
+    LogMsg "         SwitchName = ${switch_name}"
+    LogMsg "         nicsAmount = ${nicsAmount}"
+
+    # Change the working directory to where we should be
+    Set-Location $rootDir
+
+    # Verify the target VM exists, and that it is a Gen2 VM
+    LogMsg "Info : Verify the SUT VM exists"
```
Comment (chvalean) [inline Testscripts/Windows/NET-HotAddRemove-Max-NIC.ps1]: remove this

File: Testscripts/Windows/NET-HotAddRemove-Max-NIC.ps1 (hunk 16 above)
Comment (chvalean) [reply Testscripts/Windows/NET-HotAddRemove-Max-NIC.ps1]: And typo in Multiple

File: XML/TestCases/FunctionalTests.xml [hunk 30]
```diff
@@ -1330,4 +1330,54 @@
...
+        <Area>NET</Area>
+        <Tags>net</Tags>
+    </test>
+    <test>
+        <testName>NET-HOT-ADD-REMOVE-MAXNIC</testName>
+        <testScript>NET-HotAddRemove-Max-NIC.ps1</testScript>
+        <files>.\Testscripts\Linux\NET-MAX-NICs.sh,.\Testscripts\Linux\NET-Verify-HotAdd-MultiNIC.sh,.\Testscripts\Linux\utils.sh</files>
+        <setupType>SingleVM</setupType>
+        <TestParameters>
+            <param>TEST_TYPE=synthetic</param>
```
Comment (chvalean) [inline XML/TestCases/FunctionalTests.xml:1415]: param not used

File: Libraries/CommonFunctions.psm1 (hunk 1 above)
Comment (asofron) [reply Libraries/CommonFunctions.psm1]: Fixed

File: Libraries/CommonFunctions.psm1 (hunk 3 above)
Comment (asofron) [reply Libraries/CommonFunctions.psm1]: Removed

File: Libraries/CommonFunctions.psm1 (hunk 4 above)
Comment (asofron) [reply Libraries/CommonFunctions.psm1]: Changed

File: Testscripts/Linux/NET-Verify-Boot-NoNIC.sh (hunk 5 above)
Comment (asofron) [reply Testscripts/Linux/NET-Verify-Boot-NoNIC.sh]: removed

File: Testscripts/Linux/NET-Verify-Boot-NoNIC.sh (hunk 6 above)
Comment (asofron) [reply Testscripts/Linux/NET-Verify-Boot-NoNIC.sh]: removed

File: Testscripts/Linux/NET-Verify-Boot-NoNIC.sh (hunk 7 above)
Comment (asofron) [reply Testscripts/Linux/NET-Verify-Boot-NoNIC.sh:21]: Removed

File: Testscripts/Linux/NET-Verify-Boot-NoNIC.sh (hunk 8 above)
Comment (asofron) [reply Testscripts/Linux/NET-Verify-Boot-NoNIC.sh:45]: Removed

File: Testscripts/Linux/NET-Verify-HotAdd-MultiNIC.sh (hunk 9 above)
Comment (asofron) [reply Testscripts/Linux/NET-Verify-HotAdd-MultiNIC.sh:116]: Removed

File: Testscripts/Linux/NET-Verify-HotAdd-MultiNIC.sh (hunk 10 above)
Comment (asofron) [reply Testscripts/Linux/NET-Verify-HotAdd-MultiNIC.sh:13]: Removed

File: Testscripts/Windows/NET-Boot-NoNIC-HotAdd-NIC.ps1 (hunk 11 above)
Comment (asofron) [reply Testscripts/Windows/NET-Boot-NoNIC-HotAdd-NIC.ps1:234]: Added

File: Testscripts/Windows/NET-HotAdd-MultiNIC.ps1 (hunk 12 above)
Comment (asofron) [reply Testscripts/Windows/NET-HotAdd-MultiNIC.ps1:110]: Added

File: Testscripts/Windows/NET-HotAddRemove-Max-NIC.ps1 (hunk 13 above)
Comment (asofron) [reply Testscripts/Windows/NET-HotAddRemove-Max-NIC.ps1]: Fixed

File: Testscripts/Windows/NET-HotAddRemove-Max-NIC.ps1 (hunk 14 above)
Comment (asofron) [reply Testscripts/Windows/NET-HotAddRemove-Max-NIC.ps1:70]: Fixed

File: Testscripts/Windows/NET-HotAddRemove-Max-NIC.ps1 (hunk 16 above)
Comment (asofron) [reply Testscripts/Windows/NET-HotAddRemove-Max-NIC.ps1]: Fixed and fixed

File: Libraries/CommonFunctions.psm1 (hunk 17 above)
Comment (asofron) [reply Libraries/CommonFunctions.psm1]: kept

File: Testscripts/Linux/NET-Verify-HotAdd-MultiNIC.sh (hunk 19 above)
Comment (asofron) [reply Testscripts/Linux/NET-Verify-HotAdd-MultiNIC.sh]: Added

File: Testscripts/Linux/NET-Verify-HotAdd-MultiNIC.sh (hunk 20 above)
Comment (asofron) [reply Testscripts/Linux/NET-Verify-HotAdd-MultiNIC.sh]: Removed

File: Testscripts/Windows/NET-Boot-NoNIC-HotAdd-NIC.ps1 (hunk 21 above)
Comment (asofron) [reply Testscripts/Windows/NET-Boot-NoNIC-HotAdd-NIC.ps1]: Removed

File: Testscripts/Windows/NET-Boot-NoNIC-HotAdd-NIC.ps1 (hunk 23 above)
Comment (asofron) [reply Testscripts/Windows/NET-Boot-NoNIC-HotAdd-NIC.ps1]: Added

File: Testscripts/Windows/NET-Boot-NoNIC-HotAdd-NIC.ps1 (hunk 25 above)
Comment (asofron) [reply Testscripts/Windows/NET-Boot-NoNIC-HotAdd-NIC.ps1]: Updated comment to reflect action

File: Testscripts/Windows/NET-Boot-NoNIC-HotAdd-NIC.ps1 (hunk 26 above)
Comment (asofron) [reply Testscripts/Windows/NET-Boot-NoNIC-HotAdd-NIC.ps1]: Updated

File: Testscripts/Windows/NET-HotAdd-MultiNIC.ps1 (hunk 27 above)
Comment (asofron) [reply Testscripts/Windows/NET-HotAdd-MultiNIC.ps1]: Removed

File: Testscripts/Windows/NET-HotAddRemove-Max-NIC.ps1 (hunk 29 above)
Comment (asofron) [reply Testscripts/Windows/NET-HotAddRemove-Max-NIC.ps1]: Removed

File: XML/TestCases/FunctionalTests.xml (hunk 30 above)
Comment (asofron) [reply XML/TestCases/FunctionalTests.xml:1415]: removed

File: Testscripts/Windows/NET-HotAddRemove-Max-NIC.ps1 (hunk 28 above)
Comment (asofron) [reply Testscripts/Windows/NET-HotAddRemove-Max-NIC.ps1]: Removed

File: Libraries/CommonFunctions.psm1 (hunk 2 above)
Comment (asofron) [reply Libraries/CommonFunctions.psm1:4299]: moved else

Comment (asofron) [reply Libraries/CommonFunctions.psm1:4299]: Moved else

**Pull request closed without merging**